buildbot_travis/ui/src/app/configEditors/deploy.controller.coffee

```diff
@@ -138,13 +154,32 @@ class Deploy extends Controller
                                                                         else
                                                                             new_version = properties['version'][0]
                                                                             if new_version not in s.versions
-                                                                                s.versions.push(properties['version'][0])
+                                                                                s.versions.push(new_version)
                                                                 stage_info = {}
                                                                 stage_info = p.stages.toDict('stage')
                                                                 p.stages_sorted = stage_info
                                                         $scope.latestDeployedVersionByProjectDict = latestDeployedVersionByProject.toDict('projectName')
-
-
+                                            # Without a defined version property (ie commit has not been GIT tagged yet, GIT tag is generated by the deploy build)
+                                            else if properties['revision'][0]? and properties['revision'][0] != '' and properties['version'][0] == ''
+                                                if properties['stage'][0]? and properties['stage'][0] != ''
+                                                    if latestDeployedVersionByProject != []
+                                                        for p in latestDeployedVersionByProject
+                                                            if p.projectName == projectName
+                                                                for x in gitTagRevisionMapDict[projectName].map
+                                                                    if x.rev == properties['revision'][0]
+                                                                        for s in p.stages
+                                                                            if s.stage == properties['stage'][0]
```

Comment:
I am not very confortable with such 10 times indent code. Can you refactor, and make more functions?